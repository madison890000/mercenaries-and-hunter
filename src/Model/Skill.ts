import DataModel from './types';
import { v4 } from 'uuid';

export default class Skill {
    public name: string;
    public level: DataModel.SkillLevel;
    public importance: DataModel.Importance;
    public ages: number;
    id: string;

    constructor(name: string, level: DataModel.SkillLevel, ages: number, importance: DataModel.Importance) {
        this.id = v4();
        this.name = name;
        this.level = level;
        this.ages = ages;
        this.importance = importance;
    }

}
