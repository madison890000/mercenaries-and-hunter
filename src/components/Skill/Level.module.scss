.level {
  min-width: 150px;

  & > span {
    display: inline-block;
    margin: var(--base-gap-unit);
    width: 26px;
    height: 20px;
    background: var(--skill-level-fill);

    @media screen and (max-width: 1150px) {
      width: 24px;
      height: 16px;
      background: var(--skill-level-fill);
    }

    @media screen and (max-width: 414px) {
      width: 12px;
      height: 8px;
    }
  }

  @media screen and (max-width: 414px) {
    min-width: 80px;
  }


}

.levelName {
  font-size: var(--base-font-size-small);
  margin-left: var(--base-gap-unit);
}

.empty {
  width: 24px !important;
  height: 18px !important;
  background: transparent !important;
  border: 1px solid var(--skill-level-fill);


  @media screen and (max-width: 1150px) {
    width: 22px !important;
    height: 14px !important;
    border: 1px solid var(--skill-level-fill);
  }
  @media screen and (max-width: 414px) {
    width: 10px !important;
    height: 6px !important;
  }
}
