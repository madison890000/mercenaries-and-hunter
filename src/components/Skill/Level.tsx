import React from 'react';
import styles from './Level.module.scss';
import classNames from 'classnames';

const LevelUnderstand = () =>
    <div className={classNames(styles.level)}>
        <span />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <div className={styles.levelName}>Understand</div>
    </div>
const LevelFamiliar = () =>
    <div className={classNames(styles.level)}>
        <span />
        <span />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <span className={styles.empty} />
        <div className={styles.levelName}>Familiar</div>
    </div>
const LevelProficient = () =>
    <div className={classNames(styles.level)}>
        <span />
        <span />
        <span />
        <span />
        <span className={styles.empty} />
        <div className={styles.levelName}>Proficient</div>
    </div>

const Level = () => null

Level.Understand = LevelUnderstand;
Level.Familiar = LevelFamiliar;
Level.Proficient = LevelProficient;

export default Level
