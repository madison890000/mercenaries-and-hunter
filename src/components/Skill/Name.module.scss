.important {
    font-weight: bold;
}

.normal {
    color: var(--skill-name-color);
    font-weight: bold;
}

.less {
    color: var(--skill-name-color);
}
