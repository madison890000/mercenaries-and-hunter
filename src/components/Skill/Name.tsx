import React, { PropsWithChildren } from 'react';
import styles from './Name.module.scss';
import DataModel from '../../Model/types';

const Name: React.FC<PropsWithChildren<{ level: DataModel.Importance }>> = ({ level, children }) => {
    switch (level) {
        case DataModel.Importance.important:
            return <div className={styles.important}>{children}</div>;
        case DataModel.Importance.normal:
            return <div className={styles.normal}>{children}</div>;
        case DataModel.Importance.less:
            return <div className={styles.less}>{children}</div>;
        default:
            return null;
    }
};

export default Name;
