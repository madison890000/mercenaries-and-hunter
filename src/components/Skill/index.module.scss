.skill {
  display: flex;
  align-items: center;
  justify-content: space-between;

  & > span {
    margin: var(--inline-block-margin);
  }

}

.subTitle {
  font-size: var(--base-font-size-small);
  margin-left: var(--base-gap-unit);
  font-style: italic;
}

.level {
  & > span {
    display: inline-block;
    margin: var(--base-gap-unit);
    width: 36px;
    height: 20px;
    background: gray;
  }

}

.levelName {
  font-size: var(--base-font-size-small);
  margin-left: var(--base-gap-unit);
}

.understand {
  & > span:nth-child(2) {
    background: transparent;
  }

  & > span:nth-child(3) {
    background: transparent;
  }
}

.familiar {
  & > span:nth-child(3) {
    background: transparent;
  }
}

.proficient {
  color: black;
}

.important {
  font-weight: bold;
}

.normal {
  color: gray;
  font-weight: bold;
}

.less {
  color: gray;
}
