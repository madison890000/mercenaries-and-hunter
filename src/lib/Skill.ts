import { Importance, SkillLevel } from './types';
import { v4 } from 'uuid';

export default class Skill {
    public name: string;
    public level: SkillLevel;
    public importance: Importance;
    public ages: number;
    id: string;

    constructor(name: string, level: SkillLevel, ages: number, importance: Importance) {
        this.id = v4();
        this.name = name;
        this.level = level;
        this.ages = ages;
        this.importance = importance;
    }

}
