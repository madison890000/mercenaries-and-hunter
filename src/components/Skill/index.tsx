import React from 'react';
import styles from './index.module.scss';
import { Importance, SkillLevel } from '../../lib/types';
import Name from './Name';
import Level from './Level';
import { defineMessages, useIntl } from 'react-intl';

interface SkillProps {
    name: string;
    level: SkillLevel;
    ages: number;
    importance: Importance;
}

const messages = defineMessages({
    years: {
        id: 'component.skill.years',
        defaultMessage: 'years',
    },
});
const Skill = ({ level, name, ages, importance }: SkillProps) => {
    const intl = useIntl();
    return (
        <div className={styles.skill}>
            <div>
                <Name level={importance}>{name}</Name>
                <div className={styles.subTitle}>{ages} {intl.formatMessage(messages.years)}</div>
            </div>
            {level === SkillLevel.familiar && <Level.Familiar />}
            {level === SkillLevel.proficient && <Level.Proficient />}
            {level === SkillLevel.understand && <Level.Understand />}
        </div>
    );
}

export default Skill;
