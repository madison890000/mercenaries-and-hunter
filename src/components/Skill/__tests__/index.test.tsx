import React from 'react';
import Skill from '../index';
import renderer from 'react-test-renderer';
import DataModel from '../../../Model/types';

describe('renders Skill ', () => {
    test('Skill --->level = understand', () => {
        const SkillUnderstandNormal = renderer.create(<Skill name="understand normal Skill" ages={5} level={DataModel.SkillLevel.understand} importance={DataModel.Importance.normal} />).toJSON();
        expect(SkillUnderstandNormal).toMatchSnapshot();
        const SkillUnderstandLess = renderer.create(<Skill name="understand less Skill" ages={5} level={DataModel.SkillLevel.understand} importance={DataModel.Importance.less} />).toJSON();
        expect(SkillUnderstandLess).toMatchSnapshot();
        const SkillUnderstandImportant = renderer
            .create(<Skill name="understand important Skill" ages={5} level={DataModel.SkillLevel.understand} importance={DataModel.Importance.important} />)
            .toJSON();
        expect(SkillUnderstandImportant).toMatchSnapshot();
    });

    test('Skill --->level = familiar', () => {
        const SkillFamiliarNormal = renderer.create(<Skill name="familiar normal Skill" ages={5} level={DataModel.SkillLevel.familiar} importance={DataModel.Importance.normal} />).toJSON();
        expect(SkillFamiliarNormal).toMatchSnapshot();
        const SkillFamiliarLess = renderer.create(<Skill name="familiar less Skill" ages={5} level={DataModel.SkillLevel.familiar} importance={DataModel.Importance.less} />).toJSON();
        expect(SkillFamiliarLess).toMatchSnapshot();
        const SkillFamiliarImportant = renderer.create(<Skill name="familiar important Skill" ages={5} level={DataModel.SkillLevel.familiar} importance={DataModel.Importance.important} />).toJSON();
        expect(SkillFamiliarImportant).toMatchSnapshot();
    });

    test('Skill --->level = proficient', () => {
        const SkillProficientNormal = renderer.create(<Skill name="proficient normal Skill" ages={5} level={DataModel.SkillLevel.proficient} importance={DataModel.Importance.normal} />).toJSON();
        expect(SkillProficientNormal).toMatchSnapshot();
        const SkillProficientLess = renderer.create(<Skill name="proficient less Skill" ages={5} level={DataModel.SkillLevel.understand} importance={DataModel.Importance.less} />).toJSON();
        expect(SkillProficientLess).toMatchSnapshot();
        const SkillProficientImportant = renderer
            .create(<Skill name="proficient important Skill" ages={5} level={DataModel.SkillLevel.understand} importance={DataModel.Importance.important} />)
            .toJSON();
        expect(SkillProficientImportant).toMatchSnapshot();
    });
});
