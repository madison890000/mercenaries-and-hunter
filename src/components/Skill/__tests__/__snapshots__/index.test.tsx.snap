// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Skill  Skill --->level = familiar 1`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Familiar
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = familiar 2`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Familiar
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = familiar 3`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Familiar
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = proficient 1`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span />
    <span />
    <span />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Proficient
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = proficient 2`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = proficient 3`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = understand 1`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = understand 2`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = understand 3`] = `
<div
  className="skill"
>
  <div>
    <div
      className="subTitle"
    >
      5
       
      years
    </div>
  </div>
  <div
    className="level"
  >
    <span />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <span
      className="empty"
    />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;
